python/dolfin/function.py

```diff
@@ -115,13 +115,14 @@ def eval(self, x: np.ndarray, cells: np.ndarray, u=None) -> np.ndarray:
 
     def interpolate(self, u) -> None:
         """Interpolate an expression"""
-
         @singledispatch
         def _interpolate(u):
-            try:
-                self._cpp_object.interpolate(u)
-            except TypeError:
+            if isinstance(u, self.__class__):
```

Comment:
We're trying to avoid `isinstance`. Was the `try`/`except` broken?

Reply:
Yes, it threw: `Fatal Python error: Cannot recover from stack overflow.`